spec/unit/graph/simple_graph_spec.rb

```diff
@@ -681,16 +686,20 @@ def graph_to_yaml(graph, which_format)
       end
 
       it "should be able to serialize a graph where the vertices contain backreferences to the graph (#{which_format} format)" do
+        pending('Make SimpleGraph a known to the Puppet Type System')
```

Comment:
No, this test cannot run unless SimpleGraph as a PuppetObject. That's not related to `rich_data` but to the fact that the SimpleGraph object is not `Data`. I tried relying on special YAML tags here, but that fails when Ruby < 2.3 (hence my question earlier in HipChat). So if we really want this functionality, then we need to reintroduce `#to_yaml_properties` or defer the problem until later (if it is a problem).